app/controllers/index.rb

```diff
@@ -0,0 +1,155 @@
+get '/' do
+  session[:id]
+  erb :index
+end
+
+get '/browse' do
+  @all_dogs = Dog.all.sort
+  erb :browse
+end
+
+get '/search' do
+  erb :search
+end
+
+get '/search/results' do
+  @breed_list = Dog.where(breed: params[:breed])
+  p @breed_list.empty?
+  erb :search_results
+end
+
+get '/login/admin' do
+
+  erb :loginadmin
+end
+
+post '/login/admin' do
+  @admin = Admin.find_by(username: params[:username])
+  if @admin && @admin.password == params[:password]
+    session[:admin_id] = @admin.id
+  else
+    redirect '/login/admin'
+  end
+  redirect '/options'
+end
+
+get '/login' do
+  erb :login
+end
+
+post '/login' do
+  @volunteer = Volunteer.find_by(username: params[:username])
+  #if the username matches the db, then, redirect them to the main page
+  if @volunteer && @volunteer.password == params[:password]
+    session[:volunteer_id] = @volunteer.id #it stores the volunteer_id, which indicates they are logged in so they don't have to keep logging in and you don't have to keep looking their volunteer info when they go to a different page
+  else
+    @error = true
+    erb :login
+  end
+  redirect '/'
+end
+
+get '/options' do
+  @all_dogs = Dog.all.sort
+  erb :options
+end
+
+post '/options' do
+  @rate_this_dog = Dog.find(params[:id])
+
+  redirect "/options/#{params[:id]}/change"
+end
+
+get '/options/:id/change' do
+  @view_dog = Dog.find(params[:id])
+
+  erb :changeprofile
+end
+
+get '/dogs/:id/edit' do
+  @edit_dog = Dog.find(params[:id])
+
+  erb :dogs_edit
+end
+
+put '/dogs/:id/edit' do
+  @edit_dog = Dog.find(params[:dog_id])
+
+  changes = params.select {|key, value| value != ""}
+  changes.delete("_method")
+  changes.delete("dog_id")
+  changes.delete("splat")
+  changes.delete("captures")
+  changes.delete("id")
+  @edit_dog.update_attributes(changes)
```

Comment:
pull the things you want out of params rather than deleting things you dont want. White list is better than black list when it comes to params.